queries/erlang/highlights.scm

```diff
@@ -26,79 +62,68 @@
 
 [
   "catch"
-	"try"
-	"throw"
+  "try"
 ] @exception
+((atom) @exception (#any-of? @exception "error" "throw" "exit"))
+
+((atom) @boolean (#any-of? @boolean "true" "false"))
+
+;; Macros
+(macro_call_expr
+  "?" @macro
+  name: (_) @macro
+) @macro
+
+;; Preprocessor
+(pp_define) @define
+[(pp_include) (pp_include_lib)] @include
+(_preprocessor_directive) @preprocessor
+
 ;;; module define
 [
   "module"
   "export"
+  "export_type"
+  "include"
+  "include_lib"
+  "behaviour"
+  "behavior"
 ] @include
-;;; operator
-[
-  ":"
-  ":="
-  "?"
-  "!"
-  "-"
-  "+"
-  "="
-  "->"
-  "=>"
-	"|"
-	;;;TODO
-	"$"
- ] @operator
+
+;; Records
+(record_field_expr _ @structure) @structure
+(record_field_name _ @structure) @structure
+(record_name name: _ @structure) @structure
+(record_decl name: _ @structure)
+(record_field name: _ @structure ty: _ @type)
```

Comment:
```suggestion
(record_field_expr _ @type) @type
(record_field_name _ @type) @type
(record_name name: _ @type) @type
(record_decl name: _ @type)
(record_field name: _ @type ty: _ @type)
```
You could try `@type.structure`; CI should only check top-level captures but that has not worked before.

And again: why not capture the whole `(record_field_expr)` as `@type`?

Reply:
I'm still learning a bit. I think one core issue I ran into is that I tried to capture `(atom) @constant`, which seems to take priority over pretty much everything else. Also, I don't think that `@type` fits here for everything, `@field` might be more appropriate. What I want to achieve is

    A#some_record.some_field
    |- @variable
     |__________|- @type
                 |- @punctuation
                  |________|- @field

Reply:
> I'm still learning a bit. I think one core issue I ran into is that I tried to capture (atom) @constant, which seems to take priority over pretty much everything else.

Yes, but the full capture should still have priority, so the `@constant` is being "painted over" with `@type`. (Order matters here.)

> Also, I don't think that @type fits here for everything, @field might be more appropriate. What I want to achieve is

Ah, that's different; I was just referring to using the same capture inside, which should be unnecessary. `@type` is used for `structs`, which as far as I can tell is what `records` are analogues of, so they should apply to `some_record`. `some_field` is indeed different, and `@field` sounds correct for that.

Reply:
No, it's not a capture upstream, it's a highlight group -- that's different. (And that will likely be changed to `@struct`, which is what LSP semantic tokens use.)